### ISSUE TITLE
New Mobile App for SLM2

### ISSUE THREAD

**Issue by mountainmark** [sfreytag/simonlikesmaps#37]
# Orig Bug

A friend pointed me at your excellent SLM website. The android app works well on my old phones/kindle, but on my Xiaomi Redmi 10T Lite (MIUI12.5.6 Android11) the sync arrows just keep spinning.

# Plan

**Last Updated Mon 01 June 2026**

Overall Plan

- [x] Decide app runtime - PWA versus Capacitor
- [x] Set up / configure run time
- [x] Remove SSR and split code into backend / frontend
- [x] Build chosen runtime around frontend -> Capacitor
- [ ] Release workflow - fastlane / manual / appflow (dead?) / something else -> CodeMagic
    - [x] iOS to TestFlight
    - [ ] Android to Play Store Internal Testing
- [ ] Release

Mobile Fixes/Tweaks

- [ ] remove inspectable flag from testflight build
- [x] use capacitor/preferences for storing tokens
- [x] update Strava notes re token storage on app
- [x] remove rotation control from view
- [x] check OSM overlay - not working?
- [x] GPX download not working
- [x] logged out on app - tokens not renewed

Runtime bugs

- [x] Location needs Capacitor plugin
- [x] Mapbox Satellite not working - CORS error?

Capacitor runtime todo / notes

- [x] Localhost data flow - get login working
- [x] Replicate CORS config to Caddy for prod
- [x] OS maps working?
- [x] Location working?
- [x] hide tools when elev open in appmode

Remove SSR todo / notes

- [x] revisit not-found exception when done / move to caddy / delete from Nest?
- [x] create new Caddy server for frontend
- [x] build sitemap statically on client side? https://github.com/jbaubree/vite-plugin-sitemap
- [x] sort out router based on sitemap outcome
- [x] prune front/back package.json when done
- [x] once all running - remove config.isSSR / isMounted from frontend
- [x] check robots.txt ends up in right place
- [x] page titles?
- [x] hydration mismatch feature flag - needed?
- [x] compression for api?
- [x] revisit cors credentials setting
- [x] Remove cors test origin when testing complete
- [x] Test shares over HTTP (not API)

**Comment by sfreytag** [comment]
Thanks for reporting the issue, I have been away recently but now back... will reply with some thoughts soon.

**Comment by sfreytag** [comment]
The mobile app in its current form works for some people and not for others - your experience sums it up quite nicely.

The problem I've got is that the build system for it is quite old - Cordova wrapped around a cut-down version of the web app - and I've not had it up and running for a couple of years now. Even when it all worked I found the development experience a bit cumbersome for something that is ideally meant to be simple. I only got as far as Android builds and never tried an iPhone build. And then there's the barrier of getting it into the app store for Android and iPhone, which I never got around to.

All that said, I am keen to get a mobile app up and running again... I am missing using it (I have an iPhone now so cannot use the current APK). I am trying OutdoorActive but it does not do quite the same thing.

 I've opened #39 to see if a Progressive Web App will fix these issues. In theory it avoids the build overheads, works on both Android and iPhone and does not need an app store. It should fit well with the cut-down website approach.
 
 I am going to close this specific issue as a "won't fix" which realistically acknowledges that the old app won't get any more work done on it, but I did want to let you know I want to get an alternative mobile app up and running.

**Comment by sfreytag** [comment]
@mountainmark I have started a discussion to begin planning SimonLikesMaps2 which will address this mobile app problem.

Would you be interested in adding some ideas for what you'd want to see from an app? The discussion is here: [https://github.com/sfreytag/simonlikesmaps/discussions/45](https://github.com/sfreytag/simonlikesmaps/discussions/45)

**Comment by sfreytag** [comment]
Not really using the discussions so I am going to reopen this issue and make it a general 'fix the mobile app' issue.

**Comment by sfreytag** [comment]
Work on SLM2 is well underway and I have been testing Progressive Web Apps (PWA) and am fairly sure the new app plus PWA will allow me to fix all the issues with the mobile app.

**Comment by sfreytag** [comment]
SLM2 is now available for trying out - https://new.simonlikesmaps.com ... It works a lot better on a mobile and will be the basis for a mobile app. This issue is now being tracked as an item in #50 so I will close this again.

**Comment by sfreytag** [comment]
Going round in circles, but will reopen this to do after #50 and not as part of #50. This is because it is quite large, and also because the mobile app in old-SLM is not really working, so there's no need to build a new one to achieve feature-parity between old-SLM and SLM2.

**Comment by sfreytag** [comment]
Going with Capacitor, not PWA

**Comment by sfreytag** [comment]
Capacitor build progressing, app now running OK on my iPhone. Need to iron out a few usability tweaks and work out a deploy process, then try an android build. If anyone has an android phone and would like to test, let me know

**Comment by mountainmark** [comment]
Happy to try it on my Xiaomi Mi10T Lite (now on MIUI 14.0.2 - Android 12) .

**Comment by sfreytag** [comment]
> Happy to try it on my Xiaomi Mi10T Lite (now on MIUI 14.0.2 - Android 12) .

Nice to hear from you, now that I've picked this up again every time I work on it the original date of Oct 2021 catches my eye... time flies! Are you still using SLM? 

Thanks for the offer, I will keep you updated here? There's two sorts of testing I think - one just checking the app works as expected on a phone. The other will be related to installing it. The old app was an APK that you downloaded and installed yourself, needing you to toggle that one setting 'allow 3rd party apps'. I think it might be more complex nowadays with code-signing requirements. I can work thru this for the iPhone with my own phone, but if you're up for working thru that for an Android I can send you an APK earlier on and see how you get on.

**Comment by sfreytag** [comment]
Slightly slowed down with this due to unexpected contract offer in mid Jan, which has kept me busy and gets in the way of hobby projects! However, I now have SLM working in TestFlight on my iPhone. A few bugs, but the new mobile-friendly UI layout seems to work well, and work around the various safe areas / cutouts.

Next challenge is to get an Android equivalent, and then I can work on the bugs.

**Comment by sfreytag** [comment]
Location working in iOS and now implemented with a circle where the centre is your position and the radius the accuracy - ie, you are "somewhere in this circle" - useful for when GPS first acquires a signal and where accuracy can be 50m or more - you'll now notice.

**Comment by sfreytag** [comment]
All bugs fixed on iOS app, working well. A few tweaks identified, new list added.

**Comment by sfreytag** [comment]
Couple of tweaks fixed, couple more in progress... list updated.
